React/tic-tac/src/app/components/Header/index.js

```diff
@@ -1,17 +1,47 @@
 import React, { Component } from 'react';
 import logo from '@assets/logo.svg';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import actions from '@redux/auth/actions';
 
 import style from './styles.scss';
 
 class Header extends Component {
   render() {
     return (
       <header className={style.appHeader}>
-        <img src={logo} className={style.appLogo} alt="logo" />
-        <h1 className={style.appTitle}>Tic Tac Toe</h1>
+        <div className={style.child} />
+        <div className={`${style.child} ${style.childfeatured}`}>
+          <img src={logo} className={style.appLogo} alt="logo" />
+          <h1 className={style.appTitle}>Tic Tac Toe</h1>
+        </div>
+        <div className={` ${style.child} ${style.alignright}`}>
+          <ul className={style.menu}>
+            <li>
+              <button type="submit" className={style.button} onClick={this.props.logOut}>
+                LogOut
+              </button>
+            </li>
+          </ul>
+        </div>
       </header>
     );
   }
 }
 
-export default Header;
+Header.propTypes = {
+  logOut: PropTypes.func
+};
+
+const MapStateToProps = state => ({
+  loggedIn: state.auth
+});
+
+const MapDispatchToProps = dispatch => ({
+  logOut: () => dispatch(actions.logOut())
+});
+
+export default connect(
```

Comment:
Split this component into a smart or contariner component (`index.js`) and dumb (`layout.js`) component, The contariner component should only have all functional things of the component (redux things in this case), and the dumb component should only have the layout.
Follow this https://medium.com/@learnreact/container-components-c0e67432e005

Reply:
I'm not sure, if this point stayed correctly